ml-agents/mlagents/envs/rpc_communicator.py

```diff
@@ -41,20 +40,32 @@ def __init__(self, worker_id=0,
         self.server = None
         self.unity_to_external = None
         self.is_open = False
+        self.create_server()
```

Comment:
I think this is making the constructor a little bit too heavy. Is it required to create the server in the constructor ?

Reply:
Since the communicator is created in `UnityEnvironment` in the constructor and then immediately used to fetch the initial academy parameters using `initialize()`, the only other way I can see to do this is to have the `UnityEnvironment` call this method as well as the `initialize` or to combine this again with `initialize`.  Since it's not ever really safe to create the communicator without this check, it seemed to make sense to add it here.

The reasoning for separating the `create_server` from `initialize` was that they essentially do different things: `create_server` is about opening the communication whereas `initialize` is about making the initial communication.

I'm glad to change if you think there's a better way to organize this -- I'm just not seeing how the class could actually be useful before the server creation happens.  What do you think?